build_tools/github_actions/post_build_upload.py

```diff
@@ -0,0 +1,301 @@
+#!/usr/bin/env python3
+
+"""
+This script runs after building TheRock, where this script does:
+1. Create log archives
+2. Create log index files
+3. (optional) upload artifacts
+4. (optional) upload logs
+5. (optional) add links to GitHub job summary
+"""
```

Comment:
Please document here what the expected behavior is for partially complete builds / workflow runs.

This script should upload what it can even if the build failed and it should be safe to run with `if: always()` in a workflow.

You could also document what is required in the way of script inputs (including having credentials already configured in the environment for uploading)